### DISCUSSION TITLE
Testing against the Raspberry Pi Pico W

### DISCUSSION THREAD

**Discussion by rei-vilo** [phoddie/node-red-mcu#47] [Q&A] [answered]
This is the continuation of the https://github.com/phoddie/node-red-mcu/discussions/26 but more focused on the Raspberry Pi Pico W.

Why the **MCU Build Configurations** panel no longer displays the port the Raspberry Pi Pico W is connected to?

<img width="358" alt="Capture 2022-11-29 à 14 16 38" src="https://user-images.githubusercontent.com/951776/204538873-a672dbed-07d0-4ba2-83ca-19674c560870.png">

This prevents the automatic upload and the use of the debugger.

**Comment by ralphwetzel** [comment]
> Why the **MCU Build Configurations** panel no longer displays the port the Raspberry Pi Pico W is connected to?
> This prevents the automatic upload and the use of the debugger.

It does so, as the Pico is programmed via Drag&Drop to an USB drive that is setup once you connect the Pico to your system. So there's no port definition necessary.

You should see something like the following at the end of your build process:

```
Making: /Users/ralph/Projekte/Moddable/moddable/build/bin/pico/pico_w/debug/configh15lrwfhdw8/xs_pico.uf2 from xs_pico.elf
/dev/cu.usbmodem1101
Setting programming mode
product/vendor match: /dev/cu.usbmodem1101
Waiting for /Volumes/RPI-RP2...
Copying: /Users/ralph/Projekte/Moddable/moddable/build/bin/pico/pico_w/debug/configh15lrwfhdw8/xs_pico.elf to RPI-RP2
Connect to xsbug @ 2e8a:000a.
product/vendor match: /dev/cu.usbmodem1101
pico-usb-connected.
```

xsbug shall be opened & commence the debugging.

What do you get when testing this?

**Reply by rei-vilo** [reply]
Today, it works as expected: the flow is copied into **RPI-RP2**. 

Yesterday, it didn't. The **RPI-RP2** wasn't mounted.

**Comment by rei-vilo** [comment, marked as answer]
On more thing: 

The way Node-RED is launched https://github.com/phoddie/node-red-mcu/discussions/46#discussioncomment-4271902 also affects the Raspberry Pi Pico.

I installed the Pico SDK at `/Users/USER/Projects/pico` and added `export PICO_SDK_DIR=$HOME/Projects/pico/pico-sdk` to `.zshrc`.

If Node-RED is started at boot, the build process still looks at `/Users/USER/pico` and the build process fails.

Node-RED needs to be launched from a Terminal window in order to have `PICO_SDK_DIR` defined and use the correct location.

**Reply by ralphwetzel** [reply]
> If Node-RED is started at boot, the build process still looks at `/Users/USER/pico` and the build process fails.

The plugin is quite good at filling gaps in the definition of the environment variables. If you had followed the [proposed procedure](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/devices/pico.md#macos-setup) & installed the Pico SDK at the default location of `$HOME/pico/pico-sdk`, you'd ended up with a working setup.

As you decided to use a different location for the Pico SDK, it's on you to tell Node-RED where to look for those files. The easiest way to do this is by editing your `settings.js`; just make sure to keep those definitions outside of `modules.export`:

```js
process.env.PICO_SDK_DIR = "/Users/OLS/Projects/pico/pico-sdk"

module.exports = {
    ...
}

```
